src/components/table/column-picker/column-picker.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+        expect(deselected).toEqual(18);
+
+        component.deselected = ['Document ID', 'Flag', 'From', 'Icon'];
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(component.deselected.length).toEqual(4);
+        deselected = nativeElement.querySelector('.column-picker-list').querySelectorAll('.column-picker-list-item').length;
+        expect(deselected).toEqual(4);
+    });
+
+    it('should not sort the deselected columns when the sort input is undefined', () => {
+        expect(component.deselected[0]).toEqual({ group: 'Metadata', name: 'Department' });
+    });
+
+    it('should update the number of deselected columns available to select', async () => {
+        let numberOfDeselectedColumns = nativeElement.querySelector('.column-picker-stats');
+        expect(numberOfDeselectedColumns.textContent.trim()).toEqual('0 of 18 selected');
+
+        component.deselected = ['Document ID', 'Flag', 'From', 'Icon'];
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        numberOfDeselectedColumns = nativeElement.querySelector('.column-picker-stats');
+        expect(numberOfDeselectedColumns.textContent.trim()).toEqual('0 of 4 selected');
+    });
+
+    it('should update the number of select columns added', async () => {
+        let numberOfSelectedColumns = nativeElement.querySelectorAll('.column-picker-stats')[1];
+        expect(numberOfSelectedColumns.textContent.trim()).toEqual('6 columns added');
+
+        component.selected = [];
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        numberOfSelectedColumns = nativeElement.querySelectorAll('.column-picker-stats')[1];
+        expect(numberOfSelectedColumns.textContent.trim()).toEqual('1 columns added');
+    });
+});
+
+@Component({
+    selector: 'app-column-picker-sort',
+    template: `
+        <ux-column-picker
+            [(selected)]="selected"
+            [(deselected)]="deselected"
+            [groups]="groups"
+            [sort]="sort">
+        </ux-column-picker>
+    `
+})
+export class ColumnPickerSortTestComponent {
+    selected: ReadonlyArray<string | ColumnPickerGroupItem> = [
+        'Type',
+        'Date',
+        'Requested by',
+        'Status',
+        'Completion'
+    ];
+
+    deselected: Array<string | ColumnPickerGroupItem> = [
+        { group: 'Metadata', name: 'Department' },
+        { group: 'Metadata', name: 'Author' },
+        { group: 'Metadata', name: 'Category' },
+        { group: 'Metadata', name: 'Date Created' },
+        { group: 'Metadata', name: 'Date Modified' },
+        'Document ID',
+        'Flag',
+        'From',
+        'Icon',
+        'Importance',
+        'Location',
+        'Location ID',
+        'Message',
+        { group: 'Metadata', name: 'Organization' },
+        'Time',
+        'Time Created',
+        'Time Modified',
+        'Work Completed'
+    ];
```

Comment:
You can make this array a lot shorter without affecting coverage. It will make the tests simpler.